step-generation/src/__tests__/flexStackerRetrieve.test.ts

```diff
@@ -102,4 +106,41 @@ describe('flexStackerRetrieve', () => {
       python: 'wellPlate_1 = mock_flex_stacker_1.retrieve()',
     })
   })
+  it('raises an error if the hopper is empty', () => {
+    robotState = {
+      ...robotState,
+      modules: {
+        [mockModuleId]: {
+          slot: 'D3',
+          moduleState: {} as any,
+        },
+      },
+      labware: {
+        [mockLabwareId]: {
+          stack: ['D2'],
```

Comment:
nit: this isn't a valid stack, it should include the labwareId itself in it. so something like this: this means mockLabwareId is directly on the slot.


```suggestion
          stack: [mockLabwareId, 'D2'],
```